business/core/home/stores/homedb/homedb.go

```diff
@@ -0,0 +1,216 @@
+package homedb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ardanlabs/service/business/core/home"
+	db "github.com/ardanlabs/service/business/data/dbsql/pq"
+	"github.com/ardanlabs/service/business/data/order"
+	"github.com/ardanlabs/service/business/data/transaction"
+	"github.com/ardanlabs/service/foundation/logger"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+)
+
+// Store manages the set of APIs for home database access.
+type Store struct {
+	log *logger.Logger
+	db  sqlx.ExtContext
+}
+
+// NewStore constructs the api for data access.
+func NewStore(log *logger.Logger, db *sqlx.DB) *Store {
+	return &Store{
+		log: log,
+		db:  db,
+	}
+}
+
+// ExecuteUnderTransaction constructs a new Store value replacing the sqlx DB
+// value with a sqlx DB value that is currently inside a transaction.
+func (s *Store) ExecuteUnderTransaction(tx transaction.Transaction) (home.Storer, error) {
+	ec, err := db.GetExtContext(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	s = &Store{
```

Comment:
Oh, this is bad and I know you copied it. We need to fix this. I want to construct with value semantics and use the & on the return. Let's fix that everywhere.

Reply:
Okay!